src/PattyServer.js

```diff
@@ -76,18 +76,28 @@ class PattyServer {
    * @private
    */
   _setProcessOwner() {
-    // nothing to do for non-posix system
-    if (!process.setuid) { return; }
-
     // no target user to switch to
     if (!this.options.processOwner) { return; }
 
+    // set user-id
+    // nothing to do for non-posix system
+    if (!process.setuid) { return; }
     try {
       process.setuid(this.options.processOwner);
     } catch(e) {
       // throw PattyError.other()
       this.logError(`Could not change process owner to "${this.options.processOwner}"`, e);
     }
+
+    // set group-id
+    // nothing to do for non-posix system
+    if (!process.setgid) { return; }
+    try {
+      process.setgid(this.options.processOwner);
```

Comment:
Question: We assume that the user and group are the same (i.e. lke:lke)
What if the group has a different name, is there any risk?

Reply:
there is, the correct way do to this, would be to lookup the groups and check if it is in the group list.
However people usually create uid=gid at user creation.

Reply:
@yigitakbulut very good point, fixed that by getting the default group of the current user (added some more complex logic to get the group)